### ISSUE TITLE
File "location" field seem always empty

### ISSUE THREAD

**Issue by tfjmp** [End-to-end-provenance/RDataTracker#334]
Is this field supposed to contain the folder where the original can be found?  It would be really, really useful if that information was available.

**Comment by erboose** [comment]
The location field was designed to contain the full path for files read or written by the R script.  Let us know if you find examples where that is not the case.

**Comment by tfjmp** [comment]
[empty.txt](https://github.com/End-to-end-provenance/RDataTracker/files/1117937/empty.txt)

That what I assumed, but using the [graphics branch](https://github.com/End-to-end-provenance/RDataTracker/tree/graphics) it does not seem to be filled.

Looking at expected json from the tests in development it is not being filled either e.g. https://github.com/End-to-end-provenance/RDataTracker/blob/development/tests/DailySolarRadiation/expected_DailySolarRadiation_ddg.json

Same thing one master e.g. https://github.com/End-to-end-provenance/RDataTracker/blob/master/tests/DailySolarRadiation/expected_DailySolarRadiation_ddg.json

If it used to work we got a regression. When updating the test results more care must be put in verifying that they are indeed correct.

**Comment by erboose** [comment]
File location seems to have been dropped from ddg.json but not from ddg.txt.

**Comment by MKLau** [comment]
This would be great to fix if possible! 

This seems like something that Connor could tackle. @erboose Want to assign him to it?

**Comment by NullCGT** [comment]
I can work on this. I actually have some rough code to do that already built into my hashtable branch, since I needed to read the info into DDG Explorer.

**Comment by tfjmp** [comment]
That sounds great!

**Comment by NullCGT** [comment]
Would this be something that we want in an entirely new field? I'm assuming yes, since it sounds like the location field is already being used. And which branch should I do this in?

**Comment by tfjmp** [comment]
No no, the location field is not "used". It is always empty in the JSON. The idea is to actually put in the information it is supposed to contain. The field timestamp is also always empty. That is how it looks like now:
```
"d23" : {
"rdt:name" : "par-qc.csv",
"rdt:value" : "data/23-par-qc.csv",
"rdt:type" : "File",
"rdt:scope" : "undefined",
"rdt:fromEnv" : "FALSE",
"rdt:timestamp" : "",
"rdt:location" : ""
} ,
```
You may also add the file hash into the json if that is not already done. What do you think?

**Comment by NullCGT** [comment]
Sure, make sense. Would you rather I branch off of development, or do this on my hashtable branch?

**Comment by tfjmp** [comment]
I think it would be better to branch that out of development. I think it could go easily in the next release, while I am not sure what is the exact plan for hashtable.

**Comment by NullCGT** [comment]
That makes sense! I'll start on it right now.

**Comment by NullCGT** [comment]
I just fixed the locations and timestamps, I think I just have to update the tests so that Travis is happy. Afterward, I realized that I actually fixed the exact same bug in the hashtable branch a few weeks ago. Both times it took a really long time to track down, but it's a really simple one line fix where for some reason default empty string arguments are trying to be passed in during a function call.
